### PR TITLE
fix E721 types regex

### PR DESCRIPTION
**Pull request by asfaltboy**
tl;dr [The PEP8 example for the issue][0] only includes the `type(a) == type(b)` comparison, it does not include the `== types.SomeType` issue, the latter should not be handled under this error check.

This check was introduced to trunk following issue #47, and I believe it _incorrectly_ borrowed the example from the issue literally. The E721 check was since reported on a few false-positive issues and then expanded a number of times to handle those around types (see 034b9f4fb082f72e1087a8a27dd0dd61486d1888 and 0327e55e756e20ade468e67992f9f43da14aabb4).

I believe the behaviour should change because object comparison cannot be confirmed a "type comparison" in the same sense; not without having additional information, which is not possible with regex alone (_maybe_ with AST analysis).
This check also breaks many ligimitate cases (such as having your own types module with an Enum class for instance, and comparing a value to the  enum member).
If the community wants to keep it, I strongly advise it be made a separate warning level advisory (W###) rather than overloading E721, so we can handle any logic around it separately.

For reference, I learned more about the pattern, and tested for the cases using regexr:
* Original pattern - https://regexr.com/6bc9t
* Fixed pattern - https://regexr.com/6bca0

fixes issue #830

[0]: https://www.python.org/dev/peps/pep-0008/#programming-recommendations

### CODE REVIEW

File: testsuite/E72.py [hunk 1]
```diff
@@ -80,3 +78,8 @@ def func_histype(a, b, c):
     pass
 except Exception:
     pass
+#: Okay
+from . import custom_types as types
+
```
Comment (sigmavirus24) [inline testsuite/E72.py:83]: These test files aren't ever executed so we don't need a real module backing them here

Comment (asfaltboy) [reply testsuite/E72.py:83]: thanks, removed `custom_types.py` in 5df259b

**Review by sigmavirus24 — APPROVED** [review]

**Comment by asottile** [comment]
reverting this for now so I can get a release out -- this caused significant churn -- I have a proposal in #1084 to make this more palatable when reintroducing it
